Sources/tart/Fetcher.swift

```diff
@@ -3,19 +3,37 @@ import AsyncAlgorithms
 
 fileprivate let urlSession = createURLSession()
 
+class DownloadDelegate: NSObject, URLSessionTaskDelegate {
+  let progress: Progress
+  init(_ progress: Progress) throws {
+    self.progress = progress
+  }
+
+  var observation: NSKeyValueObservation?
+  func urlSession(_ session: URLSession, didCreateTask task: URLSessionTask) {
+    observation = task.progress.observe(\.fractionCompleted) { progress, value in
+      self.progress.completedUnitCount = Int64(
+        progress.fractionCompleted * Double(self.progress.totalUnitCount)
+      )
+    }
```

Comment:
Looks like it's possible to simplify this down to:

```suggestion
  func urlSession(_ session: URLSession, didCreateTask task: URLSessionTask) {
    self.progress.addChild(task.progress, withPendingUnitCount: 100)
```

Reply:
I tested that initially, but the progress of the parent progress didn't update. That's also in line with the docs at https://developer.apple.com/documentation/foundation/progress?language=objc#1661050

> The [completedUnitCount](https://developer.apple.com/documentation/foundation/progress/1407934-completedunitcount?language=objc) property for a containing progress object only updates when the suboperation is 100% complete. The [fractionCompleted](https://developer.apple.com/documentation/foundation/progress/1408579-fractioncompleted?language=objc) property for a containing progress object updates continuously as work progresses for all suboperations.

I can retry just in case I missed something in my original testing.

Reply:
>I tested that initially, but the progress of the parent progress didn't update.

Could've been because the `let downloadProgress = Progress()` was not initialized with `totalUnitCount: 100`.

> The [completedUnitCount](https://developer.apple.com/documentation/foundation/progress/1407934-completedunitcount?language=objc) property for a containing progress object only updates when the suboperation is 100% complete. The [fractionCompleted](https://developer.apple.com/documentation/foundation/progress/1408579-fractioncompleted?language=objc) property for a containing progress object updates continuously as work progresses for all suboperations.

I think this is fine because we only use the `fractionCompleted` to report the progress, see `ProgressObserver.lineToRender()`.

Reply:
You're right, thanks!

I pushed an update where I used `withPendingUnitCount: self.progress.totalUnitCount`, so that we reflect the input `Progress` fully regardless of its total count.